### PR TITLE
Update modeling_opt.py

### PR DESCRIPTION
**Pull request by sushmanthreddy**
# What does this PR do?

As suggested in the [#22561 ](https://github.com/huggingface/transformers/issues/22561) ,moved labels to the same device as logits for OTP , codegen  and gptj

@sgugger  can u pls review this once??

### CODE REVIEW

**Review by sgugger — COMMENTED** [review]
Thanks for your PR! I left a comment.
